### PR TITLE
feat: segment long plain text

### PR DESCRIPTION
**Pull request by aeng**
This change is [Reviewable](https://reviewable.io/reviews/zanata/zanata-mt/15)

### CODE REVIEW

**Comment by huangp** [comment]
:lgtm: +Reviewed +(On QA) 

---

Reviewed 4 of 6 files at r1.
Review status: 4 of 6 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/zanata/zanata-mt/15#-:-KiSVvZ4AOqeJIUzWEcJ:b9a2vem)*

**Comment by huangp** [comment]
-(Awaiting review) 

---

Review status: 4 of 6 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/zanata/zanata-mt/15#-:-KiSVyiR8hQrAGbBF1Kp:b-o9bygt)*

**Comment by huangp** [comment]
Reviewed 2 of 6 files at r1.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/zanata/zanata-mt/15)*

**Comment by huangp** [comment]
:lgtm:

---

Reviewed 5 of 5 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/zanata/zanata-mt/15#-:-Kii2p-vpX6qC7sBgOyl:bnfp4nl)*

File: src/test/java/org/zanata/mt/service/DocumentContentTranslatorServiceTest.java [hunk 1]
```diff
@@ -123,6 +125,48 @@ public void testLongHTMLCannotTranslate() {
...
+        Locale fromLocale = new Locale(LocaleId.EN, "English");
+        Locale toLocale = new Locale(LocaleId.DE, "German");
+        Document document =
+                new Document("http://localhost", fromLocale, toLocale);
+        List<String> strings = SegmentString.segmentString(text, Optional.empty());
+
+        when(persistentTranslationService.translate(document,
+                strings.subList(0, 2), fromLocale, toLocale, BackendID.MS,
+                MediaType.TEXT_PLAIN_TYPE))
+                .thenReturn(Lists.newArrayList("Translated:Hurray!", "Translated:I am never at home on Sundays. "));
```
Comment (djansen-redhat) [inline src/test/java/org/zanata/mt/service/DocumentContentTranslatorServiceTest.java]: Is this right? Should it be Hurray or Hurry?

**Comment by djansen-redhat** [comment]
✅ looks good to me

**Pull request merged**